### PR TITLE
Adds NSSecureCoding unit tests for FIRUser

### PR DESCRIPTION
**Pull request by protocol86**
Adds NSSecureCoding unit tests for FIRUser.

### CODE REVIEW

File: Example/Auth/Tests/FIRUserTests.m [hunk 1]
```diff
@@ -350,10 +352,61 @@ - (void)testUserProperties {
...
     id<FIRUserInfo> phoneUserInfo = providerMap[FIRPhoneAuthProviderID];
-    XCTAssertNotNil(phoneUserInfo);
     XCTAssertEqualObjects(phoneUserInfo.phoneNumber, kPhoneNumber);
     #endif
 
+    // Test NSSecureCoding
+    NSArray *paths =
+        NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentsDirectory = [paths objectAtIndex:0];
+    NSString *filePath = [documentsDirectory stringByAppendingPathComponent:kFilePath];
```
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRUserTests.m]: You don't need an actual file to test it. Use [initForWritingWithMutableData:](https://developer.apple.com/documentation/foundation/nskeyedarchiver/1409579-initforwritingwithmutabledata) so the tests run faster and doesn't depend on the file system.

File: Example/Auth/Tests/FIRUserTests.m [hunk 2]
```diff
@@ -350,10 +352,61 @@ - (void)testUserProperties {
     #if TARGET_OS_IOS
     // Verify FIRUserInfo properties from the phone auth provider.
     id<FIRUserInfo> phoneUserInfo = providerMap[FIRPhoneAuthProviderID];
-    XCTAssertNotNil(phoneUserInfo);
     XCTAssertEqualObjects(phoneUserInfo.phoneNumber, kPhoneNumber);
     #endif
 
+    // Test NSSecureCoding
```
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRUserTests.m]: Please also assert `[FIRUser supportsSecureCoding]` returns `YES`.

File: Example/Auth/Tests/FIRUserTests.m (hunk 1 above)
Comment (protocol86) [inline Example/Auth/Tests/FIRUserTests.m]: done, thanks!

File: Example/Auth/Tests/FIRUserTests.m (hunk 2 above)
Comment (protocol86) [inline Example/Auth/Tests/FIRUserTests.m]: added.

**Pull request merged**